contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -561,6 +564,217 @@ access(all) contract FlowTransactionSchedulerUtils {
+            } else {
+                emit COAHandlerExecutionError(id: id, owner: self.owner?.address, coaAddress: coa!.address().toString(),
+                                              errorMessage: "Invalid scheduled transaction data type for COA handler execution for tx with ID \(id)! Expected FlowTransactionSchedulerUtils.COAHandlerParams but got \(data.getType().identifier)")
+                return
+            }
+        }
+
+        /// Get the views supported by this handler
+        /// @return: Array of view types
+        access(all) view fun getViews(): [Type] {
+            return [
+                Type<COAHandlerView>(),
+                Type<StoragePath>(),
+                Type<PublicPath>(),
+                Type<MetadataViews.Display>()
+            ]
+        }
+
+        /// Resolve a view for this handler
+        /// @param viewType: The type of view to resolve
+        /// @return: The resolved view data, or nil if not supported
+        access(all) fun resolveView(_ viewType: Type): AnyStruct? {
+            if viewType == Type<COAHandlerView>() {
+                return COAHandlerView(
+                    coaOwner: self.coaCapability.borrow()?.owner?.address,
+                    coaEVMAddress: self.coaCapability.borrow()?.address(),
+                )
+            }
+            if viewType == Type<StoragePath>() {
+                return FlowTransactionSchedulerUtils.coaHandlerStoragePath()
+            } else if viewType == Type<PublicPath>() {
+                return FlowTransactionSchedulerUtils.coaHandlerPublicPath()
+            } else if viewType == Type<MetadataViews.Display>() {
+                return MetadataViews.Display(
+                    name: "COA Scheduled Transaction Handler",
+                    description: "Scheduled Transaction Handler that can execute transactions on behalf of a COA",
+                    thumbnail: MetadataViews.HTTPFile(
+                        url: ""
+                    )
+                )
+            }
+            return nil
+        }
+    }
+
+    /// Enum for COA handler execution type
+    access(all) enum COAHandlerTxType: UInt8 {
+        access(all) case DepositFLOW
+        access(all) case WithdrawFLOW
+        access(all) case Call
+
+        // TODO: Should we have other transaction types??
+    }
+
+    access(all) struct COAHandlerParams {
+
+        access(all) let txType: COAHandlerTxType
+
+        /// The amount of FLOW to deposit or withdraw
+        /// Not required for the Call transaction type
+        access(all) let amount: UFix64?
+
+        /// The following fields are only required for the Call transaction type
+        access(all) let callToEVMAddress: EVM.EVMAddress?
+        access(all) let data: [UInt8]?
+        access(all) let gasLimit: UInt64?
+        access(all) let value: EVM.Balance?   
+
+        init(txType: UInt8, amount: UFix64?, callToEVMAddress: [UInt8; 20]?, data: [UInt8]?, gasLimit: UInt64?, value: UFix64?) {
```

Comment:
IMO `callToEVMAddress` should either be `String` or `EVM.EVMAddress` - fixed byte arrays can be a pain to manipulate between txn args & this `init`. In `init` scope, you can deserialize from `String` with `EVM.addressFromString()` though FYI the method will revert on failure.

I also think `value` would benefit from `UInt` precision - UFix64 only supports 8 while EVM-native FLOW is 18 decimals. The can be some cases where that precision is needed for a successful call. I vote we either we make this `UInt?` or we accept `EVM.Balance?`

Reply:
Thanks. String is definitely a better choice there so I'll update it.

I usually try to make passing the values as arguments use simple types so people don't have to construct complex types in their app code, so I think `UInt` is a good choice for `value`. I can do that

Reply:
If I change it to `UInt`, do I just provide that to the initializer? like this?
```
self.value = EVM.Balance(attoflow: unwrappedValue)
```
I'm reaching the limits of my evm knowledge. I gotta brush up on this

Reply:
Yeah that would do it